### ISSUE TITLE
Regression: Clear validation cache removes the validators for already initiated fields

### ISSUE THREAD

**Issue by jopperknopper** [ngx-formly/ngx-formly#2786]
**Description**
We rely on schema information to have a realtime generated ecosystem of field configurations. The rendered configuration contains label names, information popups, validators and other things that are extracted from the schema.
This lets the developer to build forms with simple statements, much like building lego.

```ts
    instance.name._formly(),
    instance.surname._formly(),
    instance.role._formly(),
    instance.floodOverride._formly({
        templateOptions: {
            readOnly: true,
        }
    }),
```

Some of the custom components must verify or perform write/delete operations when switching states, for example in a tab.

When switching tabs, the validation for already initiated fields are removed. 

I have researched it and found the cause in this bugfix: **clear validation cache on fields input change #2527, closes [#2524]**

**Minimal Reproduction**
I have created a stackblitz to illustrate the problem.

https://stackblitz.com/edit/angular-adgcr6

The project is initiated with formly version 5.10.2 where the validators works.

When changing the formly version to 5.10.3, the **number field stops validating** the input.

**Environment**

- Angular version: 11
- Formly version: 5.10.15, using the material extension of formly. 

**Additional Info**
I have tried to fix this through: 
* reinitializing the form
* adding delays
* adding setTimeout() to let Angular execute the current cycle

All with negative outcome.

_A sidenote_, when adding the fields dynamically to the form, Angular always fires a ExpressionChangedAfterItHasBeenCheckedError: where the forms `valid` attribute is the culprit. 
Adding a ChangeDetectorRef.detectChanges() fixes it. But is there a deeper issue to this?

**Comment by aitboudad** [comment]
> A sidenote, when adding the fields dynamically to the form, Angular always fires a ExpressionChangedAfterItHasBeenCheckedError: where the forms valid attribute is the culprit.
Adding a ChangeDetectorRef.detectChanges() fixes it. But is there a deeper issue to this?

This is caused by `formly-form` which used in `ButtonTabsComponent` that build the fields on view render, a proper solution is to use `formly-field` instead. You may check our doc example for `tabs`: https://formly.dev/examples/advanced/tabs

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v5.10.16](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.10.16) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!

**Comment by aitboudad** [comment]
Hi, just want to mention that the above fix has been reverted in https://github.com/ngx-formly/ngx-formly/releases/tag/v5.10.27 as it introduces another [issue](https://github.com/ngx-formly/ngx-formly/issues/2965).
So the solution is to follow our example https://formly.dev/examples/advanced/tabs which use `formly-field` instead of `formly-form` in order to make it work.

Thank you!
